crates/ecs/src/filter.rs

```diff
@@ -0,0 +1,345 @@
+            );
+        };
+    }
+
+    proptest! {
+        #[test]
+        fn test_filter_identities(vars: (bool, bool, bool)) {
+            identity!(vars, With<A>, Read<A>);
+            identity!(vars, With<A>, Write<A>);
+
+            identity!(vars, Any, true);
+            identity!(vars, Not<Any>, false);
+
+            identity!(vars, Without<A>, Not<With<A>>);
+            identity!(vars, With<A>, Not<Without<A>>);
+
+            // Involution
+            identity!(vars, Not<Not<With<A>>>, With<A>);
+
+            // Dominance
+            identity!(vars, Or<With<A>, Any>, true);
+            identity!(vars, And<With<A>, Not<Any>>, false);
+
+            // Identity
+            identity!(vars, Or<With<A>, Not<Any>>, With<A>);
+            identity!(vars, And<With<A>, Any>, With<A>);
+            identity!(vars, Xor<With<A>, Not<Any>>, With<A>);
+
+            // Complementarity
+            identity!(vars, Or<With<A>, Without<A>>, true);
+            identity!(vars, And<With<A>, Without<A>>, false);
+            identity!(vars, Xor<With<A>, Without<A>>, true);
+
+            // Idempotence
+            identity!(vars, Or<With<A>, With<A>>, With<A>);
+            identity!(vars, And<With<A>, With<A>>, With<A>);
+            identity!(vars, Xor<With<A>, With<A>>, false);
+
+            // Complementarity
+            identity!(vars, Or<With<A>, Without<A>>, true);
+            identity!(vars, And<With<A>, Without<A>>, false);
+            identity!(vars, Xor<With<A>, Without<A>>, true);
+
+            // Commutativity
+            identity!(vars, Or<With<A>, With<B>>, Or<With<B>, With<A>>);
+            identity!(vars, And<With<A>, With<B>>, And<With<B>, With<A>>);
+            identity!(vars, Xor<With<A>, With<B>>, Xor<With<B>, With<A>>);
+
+            // Associativity
+            identity!(vars, Or<Or<With<A>, With<B>>, With<C>>, Or<With<A>, Or<With<B>, With<C>>>);
+            identity!(vars, And<And<With<A>, With<B>>, With<C>>, And<With<A>, And<With<B>, With<C>>>);
+            identity!(vars, Xor<Xor<With<A>, With<B>>, With<C>>, Xor<With<A>, Xor<With<B>, With<C>>>);
+
+            // Distributivity
+            identity!(vars, Or<With<A>, And<With<B>, With<C>>>, And<Or<With<A>, With<B>>, Or<With<A>, With<C>>>);
+            identity!(vars, And<With<A>, Or<With<B>, With<C>>>, Or<And<With<A>, With<B>>, And<With<A>, With<C>>>);
+
+            // Absorption
+            identity!(vars, And<With<A>, Or<With<A>, With<B>>>, With<A>);
+
+            // De Morgan's laws
+            identity!(vars, Or<With<A>, With<B>>, Not<And<Without<A>, Without<B>>>);
+            identity!(vars, And<With<A>, With<B>>, Not<Or<Without<A>, Without<B>>>);
+
+            // Xor expressed in terms of And, Or and Not
+            identity!(vars, Xor<With<A>, With<B>>, Or<And<With<A>, Without<B>>, And<Without<A>, With<B>>>);
+            identity!(vars, Xor<With<A>, With<B>>, And<Or<With<A>, With<B>>, Or<Without<A>, Without<B>>>);
+        }
```

Comment:
maybe it's implied by the tests you've already got, but it seems to me you're not testing transitivity? Might be unnecessary, but it stuck out to me as you seem to be testing all of the other properties.